### PR TITLE
Founding members form changes

### PR DESCRIPTION
**Pull request by DzhideX**
This PR aims to implement the actual functionality of the form. It still needs a bit of ironing out but for the most part it is done.

Things to be ironed out for now:
- Slow on page load as the api is being set up.
- The terms being read is sometimes not correctly figured out (resizing the screen should help).
- Other small things, possible improvements.

### CODE REVIEW

**Review by bwhm — CHANGES_REQUESTED** [review]
Good job here, almost all changes are due to sloppy communication...

## Design

### Founding Members page

#### Last period highlights

Replace the entire section, including the two tables, with a single row of "New founding members" presented in an equivalent way to the "List of current founding members", including a counter.

#### List of current founding members

The presentation of founding members, both for the purposes of the prior point, but also for the "List of current founding members", should

- Drop the bottom section for the referrer, make bottom margin equal to top margin.
- Replace "Direct score" with "Total Score"
- Replace "Referal score" with "Tokens Allocated"

#### Leaderboards

The split is now into "Founding Members" and "Non-Founding Members", not "Overall Direct Score" and "Overall Referral Score".

##### Founding Members

A table with the following columns:

 - Avatar+Handle: Custom avatar + checkmark, not membership avatar from chain. @ prefixed handle. Column does not need label.
 - Total Score
 - Tokens Allocated

##### Non-Founding Members

A table with the following columns:

 - Avatar+Handle: Membership avatar. @ prefixed handle. Column does not need label.
 - Total Score

### Leaderboard page

The split is now into "Founding Members" and "Non-Founding Members", not "Direct Score" and "Referral Score".

#### Founding Members

A table with the following columns:

 - Avatar: Custom avatar + checkmark, not membership avatar from chain. Column does not need label.
 - Handle: @ prefixed handle
 - Member ID
 - Direct Score
 - Referral Score
 - Total Score
 - Tokens Allocated
 - Inducted: Date

#### Non-Founding Members

A table with the following columns:

 - Avatar: Membership avatar. Column does not need label.
 - Handle: @ prefixed handle
 - Member ID
 - Direct Score
 - Referral Score
 - Total Score


## General

### Header

- Before the rest of pages are ready, revert header to "old" style, but add "Founding Member"
- Don't delete the work ofc, it will come back!

### Founding Members page

- With the design changes, and some other notes from below, I may have to make some changes to the json format...
- We need a consistent naming. I propose using "Founding Member" program throughout.
  - @blrhc to confirm
- "Founding members program has just launched" +...
- "What is a founding member program and why it exists?" +...
  - Highlight the fact that Jsgenesis will not be there forever?
- "Benefits" +...
- "Become a founding member" +...
  - Is this the final text here, or just something the designers made @blrhc ?
- "How to become a part of it" +...
  - Same, but also not clear what "Join now" means.
    - Is this join our telegram, like the at the bottom? (if so, rephrase)
    - Is this a link to /gettting-started? (if so, rephrase as well)
- "Full rules"will be a link to a blog, or a github repo
  - @blrhc to confirm
- "Disclaimer" missing

#### Current scoring period

- Are these values static, or read from a json?
- "Period announcement" will be a link to a blog, or a github repo?
  - @blrhc to confirm
- "How should i do it?" - add popup
- "How should I prepare it?" - add popup
  - @blrhc to write text
  - include instruction to submit via email, delivered in same format

##### Submit your summary
- As discussed, add support for password/passphrase
- Is the address matched with `root` or `controller` of membership? Should be `controller`

#### Key metrics for the program
- All data fetched from jsons
- three key metrics:
  - "size of token pool"
  - "number of founding members"
  - "number of non-founding members with 0> total score"

**Comment by blrhc** [comment]
@DzhideX 

Here's my input on all the stuff Martin has tagged me in above.

In terms of naming, it should always be Founding Member Program not Founding Members Program, and whenever referring to Founding Members it should be capitalized.

For the title section, I think it should be changed to either this one:

title and subtitle alternative

or this one:
title and subtitle

whichever you prefer!

The next section also needs the text changed:

what-is-founding-member-program


```
What is the Founding Member Program and why does it exist?

While Jsgenesis is the team currently building the technology behind the Joystream project, we will not be around forever, and we are instead entrusting the future operation of the Joystream DAO to our community members.

The Founding Member Program formally allocates JOY tokens among some of our highest quality community members in recognition of their contributions to the project and ensures that a sufficiently large, effective and motivated community of users is ready to occupy all the different roles required to run, evolve and grow the platform on mainnet.

```


Then on the next section I think it should instead be this:
benefits-section

The only changes are `special membership status ASSIGNED in genesis block` and `on official Joystream` becomes `on the official project`. Also the title bits are capitalized.

Then there are some bigger changes to the next section:
(couldn't express everything necessary in the image this time!)

how-to-participate

Instead of four steps there should be five:
1) Introduce yourself on Telegram.
2) Make contributions to the project.
3) Regularly report your activity.
4) Earn leaderboard points.
5) Become a Founding Member!

And under contributions a better list might be:
training other participants
writing blog posts and documentation
creating and publishing new video content
participating in platform roles on the testnet
finding and reporting bugs or errors
completing official bounties
creative and design work

In the next section:
`List of current founding members` -> `Current Founding Members:`

In the following section the final button should be changed:
finalbutton

The final section also needs some new text:
discuss-the-program

And the disclaimer used can probably be copied from the /token page for now.

disclaimer-section

But it should probably occupy closer to the full width of the page so it isn't so long.
My mockup is quite bad here.

---------------

Now, as far as the form section goes, one thing I noticed is that I can't drag and drop the key file, have you tested this?
Once you have selected your username and moved onto the next step, I feel as if there should be a back button on the form, otherwise it is a bit frustrating and you have to refresh the page and start again if you make a mistake.

On the second step the link to `how should i do it` needs to be `How to export your key`
The popup should say:
_In the Pioneer application [testnet.joystream.org] click the `My Keys` tab and then click the ellipsis (three dots) button on the Key which you would like to use. Then click `Create a backup file for this account`. Enter your password (if applicable) and the JSON will be downloaded._

On the third step the `How should I prepare it` link opens a popup which says:
_Your activity summary should describe in as much detail as possible your recent contributions to the success of the platform. Try to separate out each contribution into a separate paragraph or bullet point for ease of processing._

The main title of the form should change from `Submit your summary` to `Submit activity summary`

Period announcement button will provisionally link to https://github.com/Joystream/founding-member/blob/master/scoring-periods/1.md

**Comment by DzhideX** [comment]
Latest commit changes:

- Added modal before ben's comments, didn't have time to change text tonight (not styled yet as well).
- Add pasword input functionality.
- Revert navbar changes.
- Period announcement button link.
- Update disclaimer text and made it wider.

**Comment by DzhideX** [comment]
Made all the changes requested by @blrhc but:

- Didn't have time to check out and test file dropping. (Seems to be a mac/firefox thing?)
- The link for the `Read the full program rules` button.

**Comment by blrhc** [comment]
@DzhideX 

Looks like I forgot to actually paste the new text for some parts 🤦

For the hero section subtitle can the text please be: `Founding Members are highly effective project contributors awarded JOY tokens in the mainnet genesis block`?

And in the section after that, don't forget to change the text to:

`What is the Founding Member Program and why does it exist?`

Should have just written these down beforehand, quite tricky to spot in the images above, apologies.

File: src/components/FoundingMembersForm/Membership/index.js [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+              setTextInput('');
+            }
+
+            if (textInput) {
+              setMembershipHandle(textInput);
+            }
+          }
+        }}
+      />
+      <a target='_blank' href='https://github.com/Joystream/founding-members#submission-guidelines'>How it works.</a>
```
Comment (bwhm) [inline src/components/FoundingMembersForm/Membership/index.js:105]:
```suggestion
      <a target='_blank' href=https://github.com/Joystream/founding-members#submission-guidance'>How it works.</a>
```

Comment (bwhm) [reply src/components/FoundingMembersForm/Membership/index.js:105]: The url I provided was not in line with the "final" one...

File: src/data/pages/founding-members.js [hunk 2]
```diff
@@ -171,10 +172,174 @@ export const fullData = [
...
+ONBb4+CC/fKxyj7CmvVYDmL753AeVSWnzvcNuvLebrZCa8H10AIbQTfAG//hqoX5
+ZLae0VoRnU2EPG1Uh/W34+hXM46Zdk8xZt1Zl5Gkdk/1lb69Mj3RuPM9P0pFjS3N
+cqtfSrRMTi4uP5xa8b19Asnu0fkSAXnchOfW4M4GICuvgeo=
+=sbij
+-----END PGP PUBLIC KEY BLOCK-----`;
+
+export const JoystreamWSProvider = 'wss://rome-rpc-endpoint.joystream.org:9944/';
+
+export const foundingMembersJson =
+  'https://raw.githubusercontent.com/bwhm/founding-members/test-schema/data/scoring-example.json';
```
Comment (bwhm) [inline src/data/pages/founding-members.js:240]:
```suggestion
  'https://raw.githubusercontent.com/Joystream/founding-members/main/data/fm-info.json';
```

As this should be the url once [this](https://github.com/Joystream/founding-members/pull/3) is merged...

File: src/pages/founding-members/components/List/index.js [hunk 3]
```diff
@@ -1,55 +1,56 @@
...
-        <p className="FoundingMembersPage__card__score__title">Direct score</p>
-        <p className="FoundingMembersPage__card__score__stat">{score}</p>
-      </div>
-      <div className="FoundingMembersPage__card__score">
-        <p className="FoundingMembersPage__card__score__title">Referral score</p>
-        <p className="FoundingMembersPage__card__score__stat">{referralScore}</p>
+const Card = ({ founderData, partialTokenAllocation }) => {
+  const [imageHasError, setImageHasError] = useState(false);
+
+  const tokensAllocated = founderData?.extraAllocation + (founderData?.totalScore * partialTokenAllocation);
```
Comment (bwhm) [inline src/pages/founding-members/components/List/index.js:9]: I was unclear in previous comments, but there should be two values here:
```
const tokensAllocated = founderData?.extraAllocation
const tokensProjected = (founderData?.totalScore * partialTokenAllocation)
```

The former is what they have already earned, whereas the latter is only a projection. Example:

For `member1`
tokensProjected = 0.02
tokensAllocated = 3

I was initially thinking it should just be:
Tokens allocated
0.02% + 3% *

but for clarity:

Tokens allocated / projected
0.02% / 3%

**Review by bwhm — CHANGES_REQUESTED** [review]
Very small issues remaining for now.

I'm sure we'll find some small issues later, but this should hopefully be enough for launch!

File: src/pages/founding-members/components/List/index.js (hunk 3 above)
Comment (DzhideX) [reply src/pages/founding-members/components/List/index.js:9]: This is just in the List (carousels). I assume we want it in all the other places as well. Correct?

Comment (bwhm) [reply src/pages/founding-members/components/List/index.js:9]: Yes, I was just was using this as an example...

**Review by blrhc — APPROVED** [review]

**Pull request merged**